collectors/plugins.d/plugins_d.c

```diff
@@ -117,6 +117,103 @@ inline int pluginsd_split_words(char *str, char **words, int max_words) {
     return quoted_strings_splitter(str, words, max_words, pluginsd_space);
 }
 
+#ifdef ENABLE_HTTPS
+/**
+ * Update Buffer
+ *
+ * Update the temporary buffer used to parse data received from slave
+ *
+ * @param output is a pointer to the vector where I will store the data
+ * @param ssl is the connection pointer with the server
+ *
+ * @return it returns the total of bytes read on success and a negative number otherwise
+ */
+int pluginsd_update_buffer(char *output, SSL *ssl) {
+    ERR_clear_error();
+    int bytesleft = SSL_read(ssl, output, PLUGINSD_LINE_MAX_SSL_READ);
+    if(bytesleft <= 0) {
+        int sslerrno = SSL_get_error(ssl, bytesleft);
+        switch(sslerrno) {
+            case SSL_ERROR_WANT_READ:
+            case SSL_ERROR_WANT_WRITE:
+            {
+                break;
+            }
+            default:
+            {
+                u_long err;
+                char buf[256];
+                int counter = 0;
+                while ((err = ERR_get_error()) != 0) {
+                    ERR_error_string_n(err, buf, sizeof(buf));
+                    info("%d SSL Handshake error (%s) on socket %d ", counter++, ERR_error_string((long)SSL_get_error(ssl, bytesleft), NULL), ssl);
```

Comment:
```
[ 55%] Building C object CMakeFiles/netdata.dir/database/rrdfamily.c.o
In file included from /home/christopher/CLionProjects/netdata/collectors/plugins.d/../../daemon/../libnetdata/libnetdata.h:307,
                 from /home/christopher/CLionProjects/netdata/collectors/plugins.d/../../daemon/common.h:6,
                 from /home/christopher/CLionProjects/netdata/collectors/plugins.d/plugins_d.h:6,
                 from /home/christopher/CLionProjects/netdata/collectors/plugins.d/plugins_d.c:3:
/home/christopher/CLionProjects/netdata/collectors/plugins.d/plugins_d.c: In function ‘pluginsd_update_buffer’:
/home/christopher/CLionProjects/netdata/collectors/plugins.d/plugins_d.c:149:26: warning: format ‘%d’ expects argument of type ‘int’, but argument 7 has type ‘SSL *’ {aka ‘struct ssl_st *’} [-Wformat=]
  149 |                     info("%d SSL Handshake error (%s) on socket %d ", counter++, ERR_error_string((long)SSL_get_error(ssl, bytesleft), NULL), ssl);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                          ~~~
      |                                                                                                                                               |
      |                                                                                                                                               SSL * {aka struct ssl_st *}
/home/christopher/CLionProjects/netdata/collectors/plugins.d/../../daemon/../libnetdata/log/log.h:85:71: note: in definition of macro ‘info’
   85 | #define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
      |                                                                       ^~~~
/home/christopher/CLionProjects/netdata/collectors/plugins.d/plugins_d.c:149:66: note: format string is defined here
  149 |                     info("%d SSL Handshake error (%s) on socket %d ", counter++, ERR_error_string((long)SSL_get_error(ssl, bytesleft), NULL), ssl);
      |                                                                 ~^
      |                                                                  |
      |                                                                  int
```

Reply:
It was fixed with the call of SSL_get_fd().